webapp/impl/src/main/java/cloud/piranha/webapp/impl/DefaultWebApplication.java

```diff
@@ -1087,6 +1092,16 @@ public InputStream getResourceAsStream(String location) {
         return resourceManager.getResourceAsStream(location);
     }
 
+    private Stream<String> getResourcePathsImpl(String path) throws IOException, URISyntaxException {
+        URL resource = resourceManager.getResource(path);
+        if (resource == null)
+            return Stream.empty();
+        return Files.list(Paths.get(resource.toURI())).map(x -> {
```

Comment:
Indeed as you alluded to, you cannot use the Files / Path APIs here as the URI might or might not be file based